### PR TITLE
The WorldAnchorManager was putting the CreateAnchor request in a queu…

### PR DESCRIPTION
**Pull request by kmukul123**
The WorldAnchorManager was putting the CreateAnchor request in a queue but delete Anchor was immediate,

this was causing wierd issues as if you created and anchor and deleted it soon, could delete before creating and it will be a little wierd.
Fixed by putting the delete request also in the queue and after dequeueing and verified the fix

### CODE REVIEW

**Comment by HodgsonSDAS** [comment]
LGTM, although I think we should look at making successful debug log messages optional.

**Comment by thebanjomatic** [comment]
I'm not intimately familiar with the WorldAnchorStore, but is there a reason to queue the delete operation instead of just pruning the queue of any pending "Create" operations for that particular `GameObject`?  
  
It seems strange to me to do the work involved of creating the world anchor if we know that we are going to delete it.

**Comment by kmukul123** [comment]
Thank you guys, @HodgsonSDAS I agree, I will remove debug.log. @thebanjomatic You are right but that thats an optimization that can follow, it will require more testing and we have to be careful, this is just a simple bug fix which will make things work as expected however as you rightly noticed it can be improved.

**Comment by kmukul123** [comment]
@HodgsonSDAS Hi Stephen, I have signed the LCA, I also checked, the debug.Log is only in case of error now. This is my first checkin to github and I am excited about it, so if there is anything else needed from me please let me know

**Comment by HodgsonSDAS** [comment]
No worries,  Now it's up to Microsoft to accept the changes.
Some of us will likely pull it and test it out before it gets accepted.

File: Assets/HoloToolkit/Utilities/Scripts/WorldAnchorManager.cs [hunk 1]
```diff
@@ -25,8 +25,12 @@ private struct AnchorAttachmentInfo
         {
             public GameObject GameObjectToAnchor { get; set; }
             public string AnchorName { get; set; }
+            public AnchorOperation Operation { get; set; }
+        }
+        public enum AnchorOperation{
+            CREATE,
```
Comment (NeerajW) [inline Assets/HoloToolkit/Utilities/Scripts/WorldAnchorManager.cs]: Any reason why these are all caps? Please make them: Create, Delete.

File: Assets/HoloToolkit/Utilities/Scripts/WorldAnchorManager.cs [hunk 2]
```diff
@@ -125,30 +133,53 @@ public void RemoveAnchor(GameObject gameObjectToUnanchor)
...
-            string AnchorName = anchorAttachmentInfo.AnchorName;
-            GameObject gameObjectToAnchor = anchorAttachmentInfo.GameObjectToAnchor;
-
-            if (gameObjectToAnchor == null)
+            switch (anchorAttachmentInfo.Operation)
             {
-                Debug.Log("GameObject must have been destroyed before we got a chance to anchor it.");
-                return;
+                case AnchorOperation.CREATE:
+                    string AnchorName = anchorAttachmentInfo.AnchorName;
```
Comment (NeerajW) [inline Assets/HoloToolkit/Utilities/Scripts/WorldAnchorManager.cs]: Please ensure TapToPlace.cs works with your changes.

**Comment by kmukul123** [comment]
i have signed the agreement which is at https://www.docusign.net/Signing/?ti=b02e012b119040b9b9c57e32e49fd03c I am not sure why it is still saying cla required.
Thanks @NeerajW I will fix the case for Enum
I have been using the changes and  I am also using TapToPlace thats where I found the bug, its much better with the changes.

**Comment by HodgsonSDAS** [comment]
@kmukul123 please address the changes above.

**Comment by HodgsonSDAS** [comment]
@kmukul123 please address the changes above.

**Comment by HodgsonSDAS** [comment]
I'm going to try closing and opening to bump cla bot.

**Comment by kmukul123** [comment]
since the agreement now shows up in green can we push these changes?

**Comment by HodgsonSDAS** [comment]
@kmukul123, please address the changes requested by @NeerajW about the casing of the enumerations.

**Comment by HodgsonSDAS** [comment]
@NeerajW Can I go ahead and merge this and then make the updates to the enum casing?

**Comment by HodgsonSDAS** [comment]
@kmukul123, please be sure to pull the latest changes into your branch and retest. I think there's been some changes to this class since you've submitted this PR.

**Comment by HodgsonSDAS** [comment]
I'm going to close this and make the changes in https://github.com/Microsoft/HoloToolkit-Unity/pull/484

**Pull request closed without merging**